Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -399,6 +404,31 @@ bool PPCSymbolDB::LoadMap(const Core::CPUThreadGuard& guard, const std::string&
     if (name[strlen(name) - 1] == '\r')
       name[strlen(name) - 1] = 0;
 
+    // Split the current name string into separate parts, and get the object name
+    // if it exists.
+    std::string processed_name = TabsToSpaces(0, name);  // Remove tabs
+    std::vector<std::string> parts = SplitString(processed_name, ' ');
+    size_t num_parts = parts.size();
+
+    std::string object_filename_string = "";
+    std::string name_string = name; // Default to the full line
+
+    if (num_parts > 0)
+    {
+      // If the first part does not contain a left bracket (not a demangled symbol), use the first
+      // part
+      if (parts[0].find('(') == std::string::npos)
+      {
+        name_string = parts[0];
+      }
+
+      // If the last part contains a ., it has to be the object name
+      if (parts[num_parts - 1].find('.') != std::string::npos)
+      {
+        object_filename_string = parts[num_parts - 1];
+      }
+    }
```

Comment:
This logic is too simple and doesn't work in some cases (e.g. SSBB) where the name will be considered as the object:
> 8000c860 srGetAppGamename/(sr_getappname.o) 


I'm pretty sure I've seen some symbols with "." in their name.